components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -225,46 +229,113 @@ private boolean validateOAuthHeader(FullHttpRequest req)
             String[] auth = req.headers().get(HttpHeaders.AUTHORIZATION).split(" ");
             if (APIConstants.CONSUMER_KEY_SEGMENT.equals(auth[0])) {
                 String cacheKey;
+                boolean isJwtToken = false;
                 String apiKey = auth[1];
                 if (WebsocketUtil.isRemoveOAuthHeadersFromOutMessage()) {
                     req.headers().remove(HttpHeaders.AUTHORIZATION);
                 }
-                //If the key have already been validated
-                if (WebsocketUtil.isGatewayTokenCacheEnabled()) {
-                    cacheKey = WebsocketUtil.getAccessTokenCacheKey(apiKey, uri);
-                    info = WebsocketUtil.validateCache(apiKey, cacheKey);
-                    if (info != null) {
-                        infoDTO = info;
-                        return info.isAuthorized();
+
+                //Initial guess of a JWT token using the presence of a DOT.
+                if (StringUtils.isNotEmpty(apiKey) && apiKey.contains(APIConstants.DOT)) {
+                    try {
+                        org.json.JSONObject decodedHeader = new org.json.JSONObject(new String(Base64.getUrlDecoder()
+                                .decode(apiKey.split("\\.")[0]), Charset.defaultCharset()));
+                        // Check if the decoded header contains type as 'JWT'.
+                        if (APIConstants.JWT.equals(decodedHeader.getString(APIConstants.JwtTokenConstants.TOKEN_TYPE))) {
+                            isJwtToken = true;
+                            if (StringUtils.countMatches(apiKey, APIConstants.DOT) != 2) {
+                                log.debug("Invalid JWT token. The expected token format is <header.payload.signature>");
+                                throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
+                                        "Invalid JWT token");
+                            }
+                        }
+                    } catch (JSONException | IllegalArgumentException e) {
+                        isJwtToken = false;
+                        log.debug("Not a JWT token. Failed to decode the token header.", e);
                     }
```

Comment:
You can use APIUtil.isvalidJWT method for this part

Reply:
done